### PR TITLE
Add serializer for `RegExp` fields.

### PR DESCRIPTION
**Pull request by iinozemtsev**
This change is [Reviewable](https://reviewable.io/reviews/google/built_value.dart/577)

### CODE REVIEW

File: built_value/lib/serializer.dart [hunk 1]
```diff
@@ -67,6 +68,7 @@ abstract class Serializers {
           ..add(new Int64Serializer())
           ..add(new JsonObjectSerializer())
           ..add(new NumSerializer())
+          ..add(RegExpSerializer())
```
Comment (davidmorgan) [inline built_value/lib/serializer.dart]: `built_value` is still compatible with Dart 1, for a few legacy uses ... please keep the `new` :)

File: built_value/lib/src/regexp_serializer.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// license that can be found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/serializer.dart';
+
+/// Runtime type for [RegExp] private implementation.
+final _runtimeType = RegExp('').runtimeType;
+
+class RegExpSerializer implements PrimitiveSerializer<RegExp> {
+  final bool structured = false;
```
Comment (davidmorgan) [inline built_value/lib/src/regexp_serializer.dart]: Not sure what this is for?

File: built_value/lib/src/regexp_serializer.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  @override
+  Object serialize(Serializers serializers, RegExp value,
+      {FullType specifiedType = FullType.unspecified}) {
+    return value.pattern;
+  }
+
+  @override
+  RegExp deserialize(Serializers serializers, Object serialized,
+      {FullType specifiedType = FullType.unspecified}) {
+    return RegExp(serialized as String);
```
Comment (davidmorgan) [inline built_value/lib/src/regexp_serializer.dart]: Needs `new`.

File: built_value/test/regexp_serializer_test.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015, Google Inc. Please see the AUTHORS file for details.
```
Comment (davidmorgan) [inline built_value/test/regexp_serializer_test.dart]: 2019 :)

File: built_value/lib/src/regexp_serializer.dart (hunk 2 above)
Comment (iinozemtsev) [reply built_value/lib/src/regexp_serializer.dart]: That's copied from StringSerializer :) removed

**Pull request merged**